tutorials/zero-to-lamp-deploy-with-chef/index.md

```diff
@@ -0,0 +1,432 @@
+    ```
+    # gce: chef-workstation
+    echo "include_recipe 'infrastructure::sql'" >> infrastructure/recipes/default.rb
+    ```
+
+It is important to be sure that the sql cookbook is included after the compute cookbook because the sql cookbook depends on the web server ip address existing.
+
+[supermarket]: https://supermarket.chef.io/cookbooks/google-cloud
+
+## Run the infrastructure cookbook
+Now that you have created your infrastructure cookbook, run it on your Chef Workstation to configure your cloud environment.
+1. Run `chef-client` in local mode with the infrastructure cookbook as the run-list.
+    ```
+    # gce: chef-workstation
+    chef-client -z -o infrastructure
+    ```
+This may take a couple of minutes as it creates the virtual machine and Cloud SQL instance.
+
+## Create a web server cookbook
+
+For the purposes of this tutorial you can download a LAMP cookbook that can be run on Debian 9 [here][cookbook]. This cookbook is adapted from https://learn.chef.io/modules/create-a-web-app-cookbook to work within the context of chef-run and connect to an external MySQL server. Download this cookbook to your Chef Workstation and unzip it within the `cookbooks` directory that you created.
+
+```
+# gce: chef-workstation
+$USER@chef-workstation:~/chef-repo/cookbooks$ ls
+google-cloud  google-gauth  google-gcompute  google-gcontainer  google-gdns
+google-glogging  google-gsql  google-gstorage  infrastructure  lamp
+```
+
+## Configure the web server cookbook for your environment
+1. Find your Cloud SQL ip address using `gcloud sql instances list` from your original machine
+
+1. Modify the `lamp/attributes/default.rb` file to match your environment. Be sure to update `default['lamp']['database']['dbhost'] = '$CLOUD_SQL_IP'` to reference the IP of your Cloud SQL instance.
+
+## Run the cookbook to configure your web server
+Configure your web server machine for ssh access from the Chef Workstation.
+1. Authenticate the Chef Workstation machine to use Compute Engine commands
+
+```
+# gce: chef-workstation
+gcloud auth activate-service-account compute@$PROJECT_NAME.iam.gserviceaccount.com --key-file=compute.json
+```
+
+1. Connect to the `web-server`. This will force the creation of a new SSH key and upload it to the GCE instance.
+
+```
+# gce: chef-workstation
+gcloud compute ssh web-server --command='exit'
+```
+
+1. Find the ip of the web server.
+
+      gcloud compute addresses list
+1. Use the [chef-run][chefrun] command to run the `lamp` cookbook on your web server.
+    ```
+    # gce: chef-workstation
+    cd ~/chef-repo/cookbooks
+    chef-run -i ~/.ssh/google_compute_engine $USER@$WEB_SERVER_IP lamp
+    ```
+1. Verify that the cookbook completed by retrieving the content hosted by the web server
+
+
+    ```
+    # gce: chef-workstation
+    curl $WEB_SERVER_IP
+    ```
+
+Or visit $WEB_SERVER_IP in your browser. If everything is working it will display a simple html table view of your Cloud SQL database.
```

Comment:
```suggestion
Or, visit $WEB_SERVER_IP in your browser. If everything is working it will display a simple html table view of your Cloud SQL database.
```